File: packages/create2/src/module/tests/packageJson.ts
import { ScFile, ScNodeType } from '@source-craft/types';
import camelCase from 'lodash-es/camelCase';
import kebabCase from 'lodash-es/kebabCase';
import upperFirst from 'lodash-es/upperFirst';
import snakeCase from 'lodash-es/snakeCase';

export const packageJson = ({packageName}: {packageName: string;}): ScFile => ({
  type: ScNodeType.File,
  content: `{
  "name": "${packageName}-test",
  "private": true,
  "version": "0.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "TRYORAMA_LOG_LEVEL=info WASM_LOG=warn RUST_LOG=warn RUST_BACKTRACE=1 TRYORAMA_HOLOCHAIN_PATH=\\"holochain\\" ts-node src/index.ts"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@holochain/tryorama": "0.4.8",
    "@types/lodash": "^4.14.158",
    "@types/node": "^14.0.14",
    "js-base64": "^3.6.0",
    "lodash": "^4.17.19",
    "tape": "^5.0.1",
    "ts-node": "^8.10.2",
    "typescript": "4.3.5"
  }
}
`
});
    

File: crates/hc_scaffold/src/scaffold/tryorama/package_json.rs
use crate::versions::{holochain_client_version, tryorama_version};

pub fn tryorama_package_json() -> String {
    format!(
        r#"{{
  "name": "tests",
  "version": "0.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {{
    "test": "foreach -g \"**/*.test.ts\" -X \"NODE_NO_WARNINGS=1 ts-node-test #{{path}}\" --no-C"
  }},
  "author": "",
  "license": "CAL-1.0",
  "dependencies": {{
    "@msgpack/msgpack": "^2.7.0",
    "@holochain/client": "{}",
    "@holochain/tryorama": "{}",
    "foreach-cli": "^1.8.1",
    "ts-node-test": "^0.2.0"
  }},
  "type": "module"
}}"#,
        holochain_client_version(),
        tryorama_version()
    )
}
